### ISSUE TITLE
long comments vs. ignore annotations

### ISSUE THREAD

**Issue by flobee** [squizlabs/PHP_CodeSniffer#2603]
```
/**
      * Return the list of available channels.
      * 
      * .... (tvdr.de project)
      *
+     * phpcs:disable
      * id Name ;bouquet :Frequenz :Parameter          :Signalquelle:Symbolrate:VPID :APID     :TPID:CAID:SID  :NID :TID :RID
      * 30 NDR 2;ZDFmobil:674000000:B8C23D12G4M16S0T8Y0:R           :0         :673=2:674=deu@3:679 :0   :16426:8468:4097:0
+     * phpcs:enable
      *
```

Is there any way to NOT WRITE phpcs tags in the inline documentation? The checker reports issues when exceeding number of columns per line, the fixer doesnt report 'REMAINING' lines. Confusing (version 3.4.2). 

Kind regards Florian

**Comment by gsherwood** [comment]
If you don't want to put annotations into your code, you can create a custom coding standard using a ruleset.xml file. Once you create this file, you tell PHPCS to use it instead of an existing standard. More info about this file can be found here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-Ruleset

What you probably want to do is have the line length sniff ignore comment line, which you can do with a setting inside a ruleset.xml file. An example is here: https://github.com/squizlabs/PHP_CodeSniffer/wiki/Customisable-Sniff-Properties#genericfileslinelength

If you have trouble creating a custom ruleset, let me know what existing standard you are using and I can help you create one.

**Comment by flobee** [comment]
Hi, thank you for feedback. I have a ruleset.xml.
It seems `property name="ignoreComments" value="true" does it but what do other people do when describing long stings? 

A warning could be good to suppess it with -n flag
or a min/max value for "ignoreComments" 
But hiding it global, i dont like.

Kind regards Florian

**Comment by gsherwood** [comment]
> what do other people do when describing long stings?

The PSR12 standard says this:
> There MUST NOT be a hard limit on line length.
> 
> The soft limit on line length MUST be 120 characters.
> 
> Lines SHOULD NOT be longer than 80 characters; lines longer than that SHOULD be split into multiple subsequent lines of no more than 80 characters each.

So these sort of standards do not put a hard limit on line length and instead just make recommendation, which PHP uses to produce a warning. Warnings are things that should be manually reviewed in a code review to ask questions like "Is this line too long for a developer to understand?". Often, the `-n` command line argument is used to suppress warnings in CI/CD environments so they don't cause a build to fail.

Personally, I try to keep all lines within a limit, but I don't enforce this with PHPCS.

**Comment by flobee** [comment]
Alright! i keep you informed if i write something which does it. The psr stuff came after me and does often not solve my problems or is too fast with decisions i detect after ages :-D thx!
